airflow-core/src/airflow/models/deadline.py

```diff
@@ -97,11 +98,9 @@ def add_deadline(cls, deadline: Deadline, session: Session = NEW_SESSION):
         session.add(deadline)
 
 
-class DeadlineReference(Enum):
+class DeadlineReference(LoggingMixin, Enum):
```

Comment:
This class is now awfully complex as an enum, and also does what normal enums don’t generally do (contain an arbitrary value via `FIXED_DATETIME`).

Maybe we should split this into two classes like this

```python
class DefinedDeadline(Enum):
    DAGRUN_LOGICAL_DATE = ...
    DAGRUN_QUEUED_AT = ...

    def evaluate(self, **kwargs):
        ...

@dataclass
class FixedDeadline:
    at: datetime

    def evaluate(self, **kwargs):
        ...

class DeadlineReference:
    DAGRUN_LOGICAL_DATE = DefinedDeadline.DAGRUN_LOGICAL_DATE
    DAGRUN_QUEUED_AT = DefinedDeadline.DAGRUN_QUEUED_AT

    FIXED_DATETIME = FixedDeadline
```

Reply:
That's an interesting approach, thanks for the suggestion.  Let me see what I can work out along those lines.

Reply:
Let me know what you think about the new implementation.